mgmt: mcumgr: grp: img_mgmt: Fix not checking image upload size

nordicjm opened the pull request:
Fixes an issue whereby upload image size would not be checked in the first packet of an upload, which would allow an image to be uploaded until it reached the point of it being too large to fit anymore.

and

    mgmt: mcumgr: grp: img_mgmt: Fix not checking write bounds
    
    Fixes an issue whereby the data packets were not checked to ensure
    that the client has not attempted to write more data than the size
    that was provided in the original upload packet.

Fixes #60899

nordicjm committed:
mgmt: mcumgr: grp: img_mgmt: Fix not checking image upload size

Fixes an issue whereby upload image size would not be checked in
the first packet of an upload, which would allow an image to be
uploaded until it reached the point of it being too large to
fit anymore.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

de-nordic commented on subsys/mgmt/mcumgr/grp/img_mgmt/src/zephyr_img_mgmt.c:
```diff
@@ -614,30 +616,35 @@ int img_mgmt_upload_inspect(const struct img_mgmt_upload_req *req,
-				return IMG_MGMT_RET_RC_FLASH_OPEN_FAILED;
-			}
+		/* Check that the area is of sufficient size to store the new image */
+		if (req->size > fa->fa_size) {
+			IMG_MGMT_UPLOAD_ACTION_SET_RC_RSN(action,
+				img_mgmt_err_str_image_too_large);
+			flash_area_close(fa);
+			LOG_ERR("Upload too large for slot: %u > %u", req->size, fa->fa_size);
+			return IMG_MGMT_RET_RC_INVALID_IMAGE_TOO_LARGE;
```
This is good enough, although we probably need some boot API function to round down the size to actually available.
This will give us some problems because we are also able to upload already padded images.

nordicjm replied:
For the application via MCUmgr this should not be a problem because the status is in the primary image slot which will be running (assuming swap using move mode)

nordicjm committed:
mgmt: mcumgr: grp: img_mgmt: Fix not checking write bounds

Fixes an issue whereby the data packets were not checked to ensure
that the client has not attempted to write more data than the size
that was provided in the original upload packet.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
